lib/tests/tests.c

```diff
@@ -3028,6 +3028,45 @@ test_recomb_map_examples(void)
     verify_recomb_map(UINT32_MAX, 16.0, p2, r2, 7);
 }
 
+static void
+test_binary_interval_search(void)
+{
+        double values[] = {0, 10, 20, 30};
```

Comment:
Really nice, good tests. I think we just want a few tests checking what happens in boundary cases also:

- What happens when we have lists of length 0, 1 and 2?

- What happens if we have a list of all zeros?

- What if we include negative numbers also?

It might make sense to split these tests up a little bit too.

Reply:
9da4c4c